config/routes.rb

```diff
@@ -0,0 +1,13 @@
+Rails.application.routes.draw do
+  devise_for :users
+  unauthenticated :user do
+    root to: 'users#index'
+  end
+
+  authenticated :user do
+    root 'groups#index', as: :authenticated_root
+  end
+
+  resources :groups
+  resources :expenses
```

Comment:
- [ ] I would recommend adding only necessary routes or maybe there are some routes that are not used? Please use `only` or `exclude to filter the routes. You can check [here](https://stackoverflow.com/questions/13937670/rails-route-to-only-index).

Reply:
How are you doing your tests boss because everything works fine on my side

Reply:
I think this should be something optional sir because its not affecting any functionality in the app. thanks

Reply:
Hi @KayLemba,

You are right everything is working fine. But What I am saying here to allow only certain routes that you are using in your application rather than exposing all your routes.

Example:

If you only want to use the `index` and `show` routes, then you can use like this as I have mentioned below:
```ruby
resources :groups, only[:index, :show] #It will allow only index and show view to see any user. It helps to secure the application as well.

```

It comes under best practices of ruby and it is required, so I can't make it optional. Please make the changes accordingly.

Hope you understand why i put this is require changes.

Thank you! :slightly_smiling_face:

Reply:
Hello @Deepakdanger, calling resources exposes all the routes I need for the application to method and this calls only existing methods inside the controller's. This requirement is not a strict requirement as given by the project rubric and I see no need to implement something that already works the same as what I have done. Should I feel the need to extend the appliucations functionality, the route I have taken with exposing routes is more viable long-term than your suggesstion.

Again I stress. This is not a requirement, do not make it one. The application functions as required and this dialogue is simply padding on a non-issue.